cf-testd/cf-testd.c

```diff
@@ -609,16 +605,151 @@ int main(int argc, char *argv[])
         Log(LOG_LEVEL_DEBUG, "Got report file contents: %s", config->report);
     }
 
-    Log(LOG_LEVEL_INFO, "Starting server...");
+    Log(LOG_LEVEL_INFO, "Starting server at %s...", config->address);
     fflush(stdout); // for debugging startup
 
-    int r = CFTestD_StartServer(config);
-    CFTestD_ConfigDestroy(config);
+    config->ret = CFTestD_StartServer(config);
 
     /* we don't really need to do this here because the process is about the
      * terminate, but it's a good way the cleanup actually works and doesn't
      * cause a segfault or something */
     ServerTLSDeInitialize(&(config->priv_key), &(config->pub_key), &(config->ssl_ctx));
 
-    return r;
+    return NULL;
+}
+
+static void HandleSignal(int signum)
+{
+    switch (signum)
+    {
+    case SIGTERM:
+    case SIGINT:
+        // flush all logging before process ends.
+        fflush(stdout);
+        exit(EXIT_FAILURE);
+        break;
+    default:
+        break;
+    }
+}
+
+/**
+ * @param ip_str string representation of an IPv4 address (the usual one, with
+ *               4 numbers separated by dots)
+ */
+static char *IncrementIPaddress(const char *ip_str)
+{
+    struct in_addr ip;
+    int ret = inet_aton(ip_str, &ip);
+    if (ret == 0)
+    {
+        Log(LOG_LEVEL_ERR, "Failed to parse address: '%s'", ip_str);
+        return NULL;
+    }
+
+    int step = 1;
+    char *last_dot = strrchr(ip_str, '.');
+    if (strcmp(last_dot + 1, "255") == 0)
+    {
+        /* avoid the network address (ending with 0) */
+        step = 2;
+    }
+    else if (strcmp(last_dot + 1, "254") == 0)
+    {
+        /* avoid the broadcast address and the network address */
+        step = 3;
+    }
+
+    uint32_t ip_num = ntohl(ip.s_addr);
+    ip_num += step;
+    ip.s_addr = htonl(ip_num);
+
+    return xstrdup(inet_ntoa(ip));
+}
+
+int main(int argc, char *argv[])
+{
+    signal(SIGINT, HandleSignal);
+    signal(SIGTERM, HandleSignal);
+
+    Log(LOG_LEVEL_VERBOSE, "Starting cf-testd");
+    CryptoInitialize();         /* TODO: remove */
+    cfnet_init(NULL, NULL);
```

Comment:
`CryptoInitialize()` is done in `cfnet_init()` and should be safe to remove. (I know you didn't introduce this)